lib/tool_shed/webapp/frontend/src/components/pages/ManageApiKey.vue

```diff
@@ -26,15 +22,28 @@ async function copyKey() {
 const params = { encoded_user_id: "current" }
 
 async function init() {
-    apiKeyFetcher(params)
+    client
+        .GET("/api/users/{encoded_user_id}/api_key", {
+            params: {
+                path: params,
+            },
+        })
         .then(({ data }) => {
+            if (!data) {
+                throw Error("Error fetching API key")
```

Comment:
It (probably) is, I was more thinking that we might be missing out on valuable server-side error messages if we cast all errors to this generic string. Can't we just import rethrowSimple ?

Reply:
Ahh, I see your point now! Thanks for clarifying!

Reply:
AFAIK we can't import rethrowSimple because the Tool Shed 2.0 client code is now completely separated from Galaxy, but I can use the proper `error_msg` content for the error reason.

Reply:
But then we're missing out on proxy errors like gateway timeout or things like that if we assume all errors are in `err_msg`. If there's no way to share code with the Galaxy client I would copy-paste rethrowSimple (or install the galaxy client from npm ?)

Reply:
Hmm... that is true, however, it seems I kept a middleware for error handling when I was experimenting with the Tool Shed client, will that cover that case?

https://github.com/galaxyproject/galaxy/pull/18532/files#diff-266bc08bc3bd8a93c2932dda21400afe346a811fb19d093b250f5925796a6416

Certainly, the Toolshed code was the first I migrated while learning, so it can benefit from some more refactoring having the new fetcher design in mind... should we give it a try in this PR or better in a follow-up?

Reply:
The middleware approach in the TS seems fine, but we should do what Galaxy does with either rethrowSimple / errorMessageAsString. I don't think the backend is decomposed so much so that we can rely on more uniform error handling.